### DISCUSSION TITLE
Spam Video Link Detection ("Finally it's here")

### DISCUSSION THREAD

**Discussion by ThioJoe** [ThioJoe/YT-Spammer-Purge#886] [Announcements]
As a workaround for detecting comments that spam such video links, I can add the video IDs of such spam-promoted videos to SpamDomainsList.txt in the [other repo](https://github.com/ThioJoe/YT-Spam-Lists). At some point I'll add filters for the actual text, but that's going to require adding additional functionality to the script that will scan for two separate parameters instead of just scanning for "finally it's here", because many real people reference such spam comments by using that phrase.

This should mean it won't matter what changes they make to the text when spamming the links. There are also other texts I've seen. Basically it seems these are view farms that promote various videos.

Examples:
<img width="284" alt="Spam Vids 1" src="https://user-images.githubusercontent.com/12518330/179265377-60a6b0b8-1638-4c26-a941-847e376eec02.png">
<img width="207" alt="spam vids 2" src="https://user-images.githubusercontent.com/12518330/179265380-35600da3-4c80-444b-b17c-7a49ddb96a7c.png">
<img width="235" alt="spam vids 3" src="https://user-images.githubusercontent.com/12518330/179265383-ce12bf15-e4a6-4c1d-9fb5-428d9ee9168e.png">
<img width="225" alt="spam vids 4" src="https://user-images.githubusercontent.com/12518330/179265385-91024bd2-42ce-431d-a372-959c4b9a720e.png">

**Comment by sanidhyy** [comment]
👍👍

**Comment by SilverscaleClockwork** [comment]
Why not filter all youtube links with something like that: `((www\.)?youtu(be\.com|\.be)\/(watch\?v=)?).+`
Since most comments containing links are a red flag anyways.
The thing that would make sense on there, would be that the user should be able to configure a whitelist.
